src/main/java/org/mskcc/cbio/portal/scripts/ImportTabDelimData.java

```diff
@@ -277,28 +294,71 @@ public void importData(int numLines) throws IOException, DaoException {
                 genericAssayStableIdToEntityIdMap = GenericAssayMetaUtils.buildGenericAssayStableIdToEntityIdMap();
             }
 
-            int lenParts = parts.length;
+            int headerColumns = headerParts.length;
 
             String line = buf.readLine();
             while (line != null) {
+
                 ProgressMonitor.incrementCurValue();
                 ConsoleUtil.showProgress();
                 boolean recordAdded = false;
 
-                // either parse line as geneset or gene for importing into 'genetic_alteration' table
-                if (isGsvaProfile) {
-                    recordAdded = parseGenesetLine(line, lenParts, sampleStartIndex, genesetIdIndex,
-                        filteredSampleIndices, daoGeneticAlteration);
-                } else if (isGenericAssayProfile) {
-                    recordAdded = parseGenericAssayLine(line, lenParts, sampleStartIndex, genericAssayIdIndex,
-                        filteredSampleIndices, daoGeneticAlteration, genericAssayStableIdToEntityIdMap);
-                } else {
-                    recordAdded = parseLine(line, lenParts, sampleStartIndex,
-                        hugoSymbolIndex, entrezGeneIdIndex, rppaGeneRefIndex,
-                        isRppaProfile, isDiscretizedCnaProfile,
-                        daoGene,
-                        filteredSampleIndices, orderedSampleList,
-                        existingCnaEvents);
+                if (FileUtil.isInfoLine(line)) {
+                    String[] rowParts = line.split("\t", -1);
+
+                    if (rowParts.length > headerColumns && line.split("\t").length > headerColumns) {
+                        ProgressMonitor.logWarning("Ignoring line with more fields (" + rowParts.length
+                                + ") than specified in the headers(" + headerColumns + "): \n" + rowParts[0]);
+                    } else if (rowParts.length < headerColumns) {
+                        ProgressMonitor.logWarning("Ignoring line with less fields (" + rowParts.length
+                                + ") than specified in the headers(" + headerColumns + "): \n" + rowParts[0]);
```

Comment:
why warning and not error?

Reply:
update: check if second warning is really correct, or are we now introducing new lenient behavior? 
Finally: consider adding a comment on top like // legacy / backward compatibility - TODO remove / make more strict